File: component-acgschedule/src/main/java/com/rabtman/acgschedule/di/module/ScheduleDetailModule.java
package com.rabtman.acgschedule.di.module;

import com.rabtman.acgschedule.mvp.contract.ScheduleDetailContract;
import com.rabtman.acgschedule.mvp.model.ScheduleDetailModel;
import com.rabtman.common.di.scope.ActivityScope;
import dagger.Module;
import dagger.Provides;

/**
 * @author Rabtman
 */
@Module
public class ScheduleDetailModule {

  private ScheduleDetailContract.View view;

  public ScheduleDetailModule(ScheduleDetailContract.View view) {
    this.view = view;
  }

  @ActivityScope
  @Provides
  ScheduleDetailContract.View providerScheduleDetailView() {
    return this.view;
  }

  @ActivityScope
  @Provides
  ScheduleDetailContract.Model providerScheduleDetailModel(ScheduleDetailModel model) {
    return model;
  }
}

File: component-acgschedule/src/main/java/com/rabtman/acgschedule/di/module/ScheduleDetailModule.kt
package com.rabtman.acgschedule.di.module

import com.rabtman.acgschedule.mvp.contract.ScheduleDetailContract
import com.rabtman.acgschedule.mvp.model.ScheduleDetailModel
import com.rabtman.common.di.scope.ActivityScope
import dagger.Module
import dagger.Provides

/**
 * @author Rabtman
 */
@Module
class ScheduleDetailModule(private val view: ScheduleDetailContract.View) {
    @ActivityScope
    @Provides
    fun providerScheduleDetailView(): ScheduleDetailContract.View {
        return view
    }

    @ActivityScope
    @Provides
    fun providerScheduleDetailModel(model: ScheduleDetailModel): ScheduleDetailContract.Model {
        return model
    }

}